docs/learn/smart-contract-developers/deploy-up-with-lsp23.md

```diff
@@ -0,0 +1,447 @@
+
+```typescript
+const salt =
+  '0xfeedfeedfeedfeedfeedfeedfeedfeedfeedfeedfeedfeedfeedfeedfeedfeed';
+```
+
+### Instantiate the contracts
+
+Instantiate the contracts that we will be interacting with using the addresses from the previous step:
+
+```typescript
+async function main() {
+  const lsp23FactoryContract = await ethers.getContractAt(
+    LSP23FactoryArtifact.abi,
+    LSP23FactoryAddress,
+  );
+
+  const universalProfileImplementationContract = await ethers.getContractAt(
+    UniversalProfileInitArtifact.abi,
+    UniversalProfileImplementationAddress,
+  );
+}
+```
+
+### Create the LSP23 Deployment Init Structs
+
+The LSP23 Linked Contracts Factory contract uses structs as parameters to deploy the Universal Profile and its Key Manager proxies. You can find more information on these structs in the [LSP23 specification](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-23-LinkedContractsFactory.md#ilsp23linkedcontractsfactory). We will use the following structs:
+
+```typescript
+async function main() {
+  // previous code
+
+  const universalProfileInitStruct = {
+    salt,
+    fundingAmount: 0,
+    implementationContract: UniversalProfileImplementationAddress,
+    initializationCalldata:
+      universalProfileImplementationContract.interface.encodeFunctionData(
+        'initialize',
+        [LSP23PostDeploymentModuleAddress],
+      ), // this will call the initialize() function of the Universal Profile and the the LSP23PostDeploymentModuleAddress as owner
+  };
+
+  const keyManagerInitStruct = {
+    fundingAmount: 0,
+    implementationContract: LSP6KeyManagerImplementationAddress,
+    addPrimaryContractAddress: true, // this will append the primary contract address to the init calldata
+    initializationCalldata: '0xc4d66de8', // initialize() function selector ( the deployed Universal Profile address will be appended to this calldata)
+    extraInitializationParams: '0x',
+  };
+}
+```
+
+### Create the Universal Profile initialisation calldata
+
+When deploying your Universal Profile, you might want to initialise it with some data. For example, you may want to set the Universal Profile [LSP3 Metadata](../../standards/universal-profile/lsp3-profile-metadata.md), set the Universal Receiver to the Universal Profile, give some [LSP6 permissions](../../standards/universal-profile/lsp6-key-manager#permissions.md) to some controllers, etc. For the following guide we will:
+
+- set the Universal Profile LSP3 Metadata
+- set the [Universal Receiver](../../standards/generic-standards/lsp1-universal-receiver.md) to the Universal Profile
+- give some permissions to the Universal Receiver ([`SUPER_SETDATA` and `REENTRANCY`](../../standards/universal-profile/lsp6-key-manager#permissions))
+- give all permissions to the main controller
+  set the [`AddressPermissions[]`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#addresspermissions) length data key
```

Comment:
I would suggest to keep these links to _docs.lukso.tech_ instead of the LIP repository, as the two previous ones link to _doc.lukso.tech_ > **Standard** pages.

This will avoid the reader to have to switch tabs and stay in the same website.

Reply:
changed some of them but i think some of them are better explained in the LIP repo